app/upload/run.go

```diff
@@ -97,16 +107,39 @@ func (uc *UpCmd) finishing(ctx context.Context) error {
 		return nil
 	}
 	defer func() { uc.finished = true }()
-	// do waiting operations
-	uc.albumsCache.Close()
-	uc.tagsCache.Close()
+
+	if uc.DeferTags {
+		if uc.client.PauseImmichBackgroundJobs {
+			uc.app.Log().Info("Resuming metadata extraction...")
+			// Use a background context to ensure the command is sent even if the main context is cancelling
+			bgCtx := context.Background()
+			_, err := uc.client.AdminImmich.SendJobCommand(bgCtx, "metadataExtraction", "resume", true)
```

Comment:
The linter doesn't like it
  Error: app/upload/run.go:116:50: Non-inherited new context, use function like `context.WithXXX` instead (contextcheck)
  			_, err := uc.client.AdminImmich.SendJobCommand(bgCtx, "metadataExtraction", "resume", true)

anyway, I don't know how to do something else


```suggestion
			_, err := uc.client.AdminImmich.SendJobCommand(bgCtx, "metadataExtraction", "resume", true) //nolint:contextcheck
```

Reply:
Maybe this is different than the original problem you are describing.

I can confirm task 2 does not override the values of task 3. I believe task 2 will simply not add tags read from the metadata if tags have already been applied to the asset.

Running the metadata task first was my way of trying to speed up the process so the tags get applied in immich and immich-go can than apply its tags as soon as possible so the user can than exit the program.

I have now tested this solution a couple times with my large library and it does work but in the event that the import gets stuck or immich tasks deadlock you end up having to reimport the whole library so writing the tags to a file first so it can always pick up would make this implementation more reliable..

That all being said I think I may investigate the immich Code base first to see if the process of applying tags could be updated there as this is turning into a large work around for what may just be a simple PR on that end.

I have also never written in Go until now so this is not my area of expertise and I don't like committing spaghetti or AI slop so I will get back to this in a bit.

Reply:
Immich used to have (has) issues with upload and modifying metadata using the API.  

I think a good suspect it the "ReadXMP" job.   
#1245  
#1200

Reply:
Haven't forgot about this but also haven't looked into it yet. If I forget about this entirely feel free to close but its the holidays so just a little busy